drivers/storage/portworx/manifest/manifest.go

```diff
@@ -312,6 +307,35 @@ func fillPrometheusDefaults(
 	}
 }
 
+func fillTelemetryDefaults(
+	rel *Version,
+) {
+	pxVersion, err := version.NewSemver(rel.PortworxVersion)
+	if err != nil || !pxutil.IsCCMGoSupported(pxVersion) {
+		// Old CCM Java telemetry
+		if rel.Components.Telemetry == "" {
+			rel.Components.Telemetry = defaultCCMJavaImage
+		}
+		if rel.Components.MetricsCollector == "" {
```

Comment:
We don't need to put all the images in if-else statements, just the `Telemetry` image.
```
if rel.Components.Telemetry == "" {
  pxVersion, err := version.NewSemver(rel.PortworxVersion)
  if err != nil || !pxutil.IsCCMGoSupported(pxVersion) {		
    rel.Components.Telemetry = defaultCCMJavaImage
  } else {
    rel.Components.Telemetry = defaultCCMGoImage
  }
}

if rel.Components.MetricsCollector == "" {
  ...
}

...

Reply:
better not to set the rest images if old/new ccm is not enabled to be safe, though we expect everything from manifest is correct. for now it should be ok.